### PR TITLE
Support for channels

### LINKED ISSUE #14: Request: Add support for channels
As an NLog-consuming program author, I would like to leverage the new features (starting with ETW channel support) available in the [Microsoft EventSource Library](https://www.nuget.org/packages/Microsoft.Diagnostics.Tracing.EventSource) so that I may later register, then use Windows Event Viewer to easily view my application's log entries as well as enable/disable channel-related logs.

### PR DESCRIPTION
**Pull request by mfaulkinbury**
Hello,
Please consider the issue that I just opened and the related pull request.  
I've added additional information to the readme.md file.
I'm looking forward to your feedback and an opportunity to contribute to this great repo.

Best,
Mike

fixes #14

### CODE REVIEW

File: NLog.Etw/NLogEtwExtendedTarget.cs [hunk 1]
```diff
@@ -1,48 +1,45 @@
...
+    /// A NLog target with support for channel-enabled ETW tracing. When using perfview or wpr to record the events use *NLog-LogEvents
     /// to enable the NLog provider.
     /// 
-    /// Sample configuration and usage sample can be found on my blog: http://lowleveldesign.wordpress.com/2014/04/18/etw-providers-for-nlog/
+    /// Channel alignment based on best practices documented here: https://blogs.msdn.microsoft.com/vancem/2012/08/14/etw-in-c-controlling-which-events-get-logged-in-an-system-diagnostics-tracing-eventsource/ 
     /// </summary>
     [Target("ExtendedEventTracing")]
     public sealed class NLogEtwExtendedTarget : TargetWithLayout
     {
-        [EventSource(Name = "LowLevelDesign-NLogEtwSource")]
```
Comment (304NotModified) [inline NLog.Etw/NLogEtwExtendedTarget.cs:18]: why this change? It's a breaking change?

File: NLog.Etw/RegisterEvents/ReadMe.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+﻿RegisteredEvents.EXE is a seld-extracting, self-installing package used to register the NLog ETW provider and channel-based log files.
```
Comment (304NotModified) [inline NLog.Etw/RegisterEvents/ReadMe.txt]: do we need these files? 

readme.txt, RegisterEvents.EXE,  /RegisterEvents.SED, _EventRegisterUsersGuide.docx .bat files etc

**Review by 304NotModified — CHANGES_REQUESTED** [review]

File: NLog.Etw/NLogEtwExtendedTarget.cs (hunk 1 above)
Comment (mfaulkinbury) [reply NLog.Etw/NLogEtwExtendedTarget.cs:18]: From [this article](https://blogs.msdn.microsoft.com/vancem/2012/08/14/etw-in-c-controlling-which-events-get-logged-in-an-system-diagnostics-tracing-eventsource/):
"Finally, the EventSource above follows a ‘best practice’ of naming its ETW provider a good name FROM THE U SER’s POINT OF VIEW.   Names should at least start with your company name.  If you expect more than one EventSource for your company it is best to give them categories separated by ‘-‘.   This is the convention that Microsoft has been following for the last several years.  (e.g. Microsoft-Windows-DotNetRuntime). "

"LowLevelDesign-xxx" doesn't seem to be a "good name from the user's point of view" (it's not obvious what that refers to).  "NLog-xxx" seems more appropriate IMHO.

It's only a breaking change with respect to guidance.  We could add updated instructions to the readme file on how to dynamically consume the events using PerfView.

Thoughts?

File: NLog.Etw/RegisterEvents/ReadMe.txt (hunk 2 above)
Comment (mfaulkinbury) [reply NLog.Etw/RegisterEvents/ReadMe.txt]: _EventRegisterUsersGuide.docx gets added with [Microsoft.Diagnostics.Tracing.EventRegister](https://www.nuget.org/packages/Microsoft.Diagnostics.Tracing.EventRegister/).  This package creates a new manifest by inspecting the event source during build.

The rest of the files are my attempt to make manifest registration easier.  I'm open to refactoring as needed or moving into a separate package if they seem to add too much friction.

File: NLog.Etw/packages.config [hunk 3]
```diff
@@ -1,4 +1,5 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
+  <package id="Microsoft.Diagnostics.Tracing.EventSource.Redist" version="1.1.28" targetFramework="net45" />
```
Comment (304NotModified) [inline NLog.Etw/packages.config:3]: The nuspec should be updated for this?

File: README.md [hunk 4]
```diff
@@ -7,3 +7,61 @@ NLog.Etw
...
+      <logger name="*" minlevel="Trace" writeTo="eetw" />
+    </rules>
+</nlog>
+```
+
+#### Register Events (optional)
+
+If you desire to leverage the Windows Event Viewer (and channel-related Event Logs), you may install the event manifest on the target machine(s).
+1. Copy RegisterEvents.exe to the target machine.
+2. Run RegisterEvents.exe as Administrator.
```
Comment (304NotModified) [inline README.md]: This is old I guess?

**Review by 304NotModified — CHANGES_REQUESTED** [review]

**Comment by 304NotModified** [comment]
Is this change backwards compatible?

**Comment by 304NotModified** [comment]
Otherwise we could add a new target instead of changing the expended

**Comment by mfaulkinbury** [comment]
This should be backward compatible (same NLog config leads to same output/ETW session).

File: NLog.Etw/packages.config (hunk 3 above)
Comment (mfaulkinbury) [reply NLog.Etw/packages.config:3]: NuGet pack was failing due to the content files reference. The package dependencies (NLog, Microsoft.Diagnostics.Tracing.EventSource.Redist) are added automagically.

**Comment by mfaulkinbury** [comment]
I'd be fine with adding a new target as well. Your call.

Sent from my mobile phone

**Comment by 304NotModified** [comment]
OK thx. No need for new target of backwards comp :)

**Review by 304NotModified — APPROVED** [review]

**Comment by 304NotModified** [comment]
Releasing this one ASAP as NLog.ETW 4.1?

**Comment by mfaulkinbury** [comment]
Were you going to push to NuGet?

**Comment by mfaulkinbury** [comment]
Sounds good to me

Sent from my mobile phone

**Pull request merged**